discovery_engine_core/bindings/src/types/migration_search.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+//! Provides bindings to `crate::storage2::Search` renamed to `MigrationSearch`.
+//!
+//! The renaming is necessary to avoid ffi naming collisions. In the ffi this
+//! type is only used for the dart->rust migration.
+
+use std::ptr::addr_of_mut;
+
+use xayn_discovery_engine_core::storage2::{Search, SearchBy};
+
+//cbindgen:ignore
+pub type MigrationSearch = Search;
+
+/// Initializes a `Option<MigrationSearch>` to `None` at given place.
+///
+/// # Safety
+///
+/// It must be sound to write any sound `Option<MigrationSearch>` to given place.
+#[no_mangle]
+pub unsafe extern "C" fn init_option_migration_search_none_at(place: *mut Option<MigrationSearch>) {
+    unsafe {
+        place.write(None);
+    }
+}
+
+/// Initializes a `Option<MigrationSearch>` to `Some(search)` at given place.
+///
+/// The boxed search is moved into this function.
+///
+/// # Safety
+///
+/// It must be sound to write any sound `Option<MigrationSearch>` to given place.
+#[no_mangle]
+pub unsafe extern "C" fn init_option_migration_search_some_at(
+    place: *mut Option<MigrationSearch>,
+    search: Box<MigrationSearch>,
+) {
+    unsafe {
+        place.write(Some(*search));
+    }
+}
+
+/// Returns a pointer to the `search_by` field of a [`MigrationSearch`] memory object.
+///
+/// # Safety
+///
+/// The pointer must point to a valid [`MigrationSearch`] memory object, it
+/// might be uninitialized.
+#[no_mangle]
+pub unsafe extern "C" fn migration_search_place_of_search_by(
+    place: *mut MigrationSearch,
+) -> *mut SearchBy {
+    unsafe { addr_of_mut!((*place).search_by) }
+}
+
+/// Returns a pointer to the `search_term` field of a [`MigrationSearch`] memory object.
+///
+/// # Safety
+///
+/// The pointer must point to a valid [`MigrationSearch`] memory object, it
+/// might be uninitialized.
+#[no_mangle]
+pub unsafe extern "C" fn migration_search_place_of_search_term(
+    place: *mut MigrationSearch,
+) -> *mut String {
+    unsafe { addr_of_mut!((*place).search_term) }
+}
+
+/// Returns a pointer to the `page_size` field of a `Paging` memory object splatted in a [`MigrationSearch`] memory object.
```

Comment:
:art: 
```suggestion
/// Returns a pointer to the `size` field of a `Paging` memory object splatted in a [`MigrationSearch`] memory object.
```
what do you mean by splatted? split?

Reply:
no splatted, word stem splat 

as in: layout in in memory in a sequential manner without any indirection, I will change it as splatting is a bit more appropriate if the position of the memory object in memory is arbitrary and there are no gaps in the data sequence